### PR TITLE
Skin attachment works in editor

### PR DESCRIPTION
**Pull request by amzn-rhhong**
https://user-images.githubusercontent.com/69218254/167970469-39514881-5bf3-4b95-9732-de884ff268c5.mp4
<-- see video above to see it in action -->

Skin attachment works in editor
image

P.S: We might be overdue on refactor ActorComponent / EditorActorComponent to use the shared component base. If we had done that work a lot of the same logic between the editor/runtime component could share the same code and avoid a lot of code duplication. 

Signed-off-by: rhhong <rhhong@amazon.com>

### CODE REVIEW

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp
Comment (hultonha) [inline Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp:1053]: nit/suggestion: We could move this inside the `if`

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp
Comment (hultonha) [inline Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp:1063]: nit/suggestion: We could use an `if` initializer here

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp
Comment (hultonha) [inline Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp:1062]: nit: consider naming this `attachmentIt` to make it a bit clear what it is referring to

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.h
Comment (hultonha) [inline Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.h:145]: nit: Missing `overrides ...` (also I'd omit `::Handler` too)

**Review by hultonha — COMMENTED** [review]
Change looks good to me but I'll defer to @amzn-jillich to have a look before I approve. I have a few small comments but nothing major. We should definitely add the refactor task you mention to the high-priority backlog too (but before that we should add some tests to codify the behavior and help avoid any breaks when we update the code). Might be worth thinking about how we can test some of this behavior to avoid future regressions (maybe consider a follow-up PR or we can create a separate task for this).

**Review by amzn-jillich — APPROVED** [review]
Looks good to me other than the fact you mentioned already that we need to refactor the actor component with its editor component as the amount of duplicated code just increases over time.

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp:1053]: Yeah I know about the c++17 if initializer thing. But somehow my brain still tell me to walk one stair at a time. Perhaps I will get use to that one day.

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp:1062]: ok i will change this one.

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.h
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.h:145]: will change

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp
Comment (hultonha) [reply Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp:1053]: I only like it as it reduces the scope so you can't accidentally misuse if `attachedTo`, `if` initializer can be nice for this as there are actually two statements instead of the implicit null check that happens without that

**Review by hultonha — APPROVED** [review]

**Pull request merged**
